pkg/util/fs_util.go

```diff
@@ -677,3 +672,49 @@ func CreateTargetTarfile(tarpath string) (*os.File, error) {
 	return os.Create(tarpath)
 
 }
+
+// Returns true if a file is a symlink
+func IsSymlink(fi os.FileInfo) bool {
+	return fi.Mode()&os.ModeSymlink != 0
+}
+
+var ErrNotSymLink = fmt.Errorf("not a symlink")
+
+func CanonicalizeLink(path string) (string, error) {
+	fi, err := os.Lstat(path)
+	if err != nil {
+		return "", err
+	}
+	if !IsSymlink(fi) {
+		return "", ErrNotSymLink
+	}
+	return filepath.EvalSymlinks(path)
+}
+
+// otiai10Cpy.Copy in case the src file is a symlink, will copy the target
+// file at destination instead of creating a symlink. See #915 for more details.
```

Comment:
I think updating this comment with more detail instead of a reference to the issue would be great.

Something like
```
// For cross stage dependencies kaniko must persist the referenced path so that it can be used in 
// the dependent stage. For symlinks we copy the target path because copying the symlink would 
// result in a dead link
```